lib/quiet_quality/executors/pipeline.rb

```diff
@@ -72,6 +69,23 @@ def relevance_filter
       def locator
         @_locator ||= AnnotationLocator.new(changed_files: changed_files)
       end
+
+      def filtered(messages_object)
+        if filter_messages? && changed_files
+          original_count = messages_object.count
+          messages_object = relevance_filter.filter(messages_object)
+          info("Messages for #{tool_name} filtered from #{original_count} to #{messages_object.count}")
+        end
+        messages_object
+      end
```

Comment:
I don't love that we're clobbering the original object in here. Maybe if it was changed to an early return we wouldn't need to do so?

```suggestion
      def filtered(messages_object)
        return messages_object unless filter_messages? && changed_files

        original_count = messages_object.count
        relevance_filter.filter(messages_object).tap do |filtered|
          info("Messages for #{tool_name} filtered from #{original_count} to #{filtered.count}")
        end
      end
```

Reply:
Hm - I regard it as "reassignment", and I think it's fairly normal. But I do try to avoid it usually - it was the simplest of the structures I tried out, but I'm sure I can make something nicer.

Reply:
Yeah, I guess early return and then a different variable name works well also.